### PR TITLE
fix: curation

### PR DESCRIPTION
**Pull request by florian-hoenicke**
fixed issues:
- persistency
- expected query field name did not match bff query field name
- update did not overwrite previous curations

### CODE REVIEW

File: now/executor/indexer/elastic/elastic_indexer.py [hunk 1]
```diff
@@ -426,25 +441,40 @@ def curate(self, parameters: dict = {}, **kwargs):
...
-                self.es.indices.refresh(index=self.index_name)
                 ids = [r['_id'] for r in resp['hits']['hits']]
-                self.query_to_curated_ids[query] += [
-                    id for id in ids if id not in self.query_to_curated_ids[query]
-                ]
+                self.query_to_curated_ids[query] = remove_duplicates(
+                    self.query_to_curated_ids.get(query, []) + ids
+                )
+
+        self.save_curated(self.query_to_curated_ids)
```
Comment (joschkabraun) [inline now/executor/indexer/elastic/elastic_indexer.py:464]: I think we can replace this function call with the code itself as it isn't reused anywhere and not doing anything original + 2 lines

File: now/executor/indexer/elastic/es_query_building.py [hunk 2]
```diff
@@ -94,6 +94,8 @@ def build_es_queries(
                 docs[doc.id] = doc
                 docs[doc.id].tags['embeddings'] = {}
 
+            print('# doc.id', doc.id)
+            print('# queries', queries)
```
Comment (joschkabraun) [inline now/executor/indexer/elastic/es_query_building.py:98]: remove print

File: now/executor/indexer/elastic/es_query_building.py [hunk 3]
```diff
@@ -102,10 +104,14 @@ def build_es_queries(
                     custom_bm25_query,
                     filter,
                 )
+                print(
+                    f'# create pinned queries for {doc.id}, {doc}, {query_to_curated_ids}'
+                )
```
Comment (joschkabraun) [inline now/executor/indexer/elastic/es_query_building.py:109]: remove print

File: now/executor/indexer/elastic/es_query_building.py [hunk 4]
```diff
@@ -102,10 +104,14 @@ def build_es_queries(
...
                 pinned_queries[doc.id] = get_pinned_query(
                     doc,
                     query_to_curated_ids,
                 )
+                print(f'# pinned_queries[{doc.id}]', pinned_queries[doc.id])
```
Comment (joschkabraun) [inline now/executor/indexer/elastic/es_query_building.py:114]: remove print

File: now/executor/indexer/elastic/es_query_building.py [hunk 5]
```diff
@@ -195,12 +201,18 @@ def get_default_query(
...
+    for c in doc.chunks:
+        curated_ids += query_to_curated_ids.get(c.text, [])
+    curated_ids = remove_duplicates(curated_ids)
+    if curated_ids:
+        return {'pinned': {'ids': curated_ids}}
+    else:
+        return {}
+
+
+def remove_duplicates(l):
```
Comment (joschkabraun) [inline now/executor/indexer/elastic/es_query_building.py:214]: let's remove this function

Comment (joschkabraun) [reply now/executor/indexer/elastic/es_query_building.py:214]: also isn't something like `list(set(l))` more conventional?

**Review by joschkabraun — CHANGES_REQUESTED** [review]

File: now/executor/indexer/elastic/es_query_building.py (hunk 5 above)
Comment (florian-hoenicke) [reply now/executor/indexer/elastic/es_query_building.py:214]: The order has to be maintained. So list(set(l)) wont work.
The function is called twice and therefore, this code would be duplicated

**Pull request merged**